src/_pytest/raises.py

```diff
@@ -704,10 +704,11 @@ def __exit__(
         if exc_type is None:
             if not self.expected_exceptions:
                 fail("DID NOT RAISE any exception")
-            if len(self.expected_exceptions) > 1:
-                fail(f"DID NOT RAISE any of {self.expected_exceptions!r}")
-
-            fail(f"DID NOT RAISE {self.expected_exceptions[0]!r}")
+            if len(self.expected_exceptions) == 1:
+                fail(f"DID NOT RAISE {self.expected_exceptions[0].__name__!r}")
+            else:
+                names = ", ".join(f"{x.__name__!r}" for x in self.expected_exceptions)
+                fail(f"DID NOT RAISE any of ({names})")
```

Comment:
Nice improvement overall! However, I think the quotes here are rather confusing, since those are classes/types and not strings. I think they made sense in the `<class 'ValueError'>` case, but as a plain name, I'd much prefer just having `ValueError`:


```suggestion
            if len(self.expected_exceptions) == 1:
                fail(f"DID NOT RAISE {self.expected_exceptions[0].__name__}")
            else:
                names = ", ".join(x.__name__ for x in self.expected_exceptions)
                fail(f"DID NOT RAISE any of ({names})")
```